src/cata_tiles.cpp

```diff
@@ -1228,13 +1327,25 @@ bool cata_tiles::draw_sprite_at(const std::vector<int> &spritelist, int x, int y
     destination.w = tile_width;
     destination.h = tile_height;
 
+    if ( svlist.empty() ) {
+        // render nothing
+        return true;
+    }
+
+    const std::vector<int> *spritelist = svlist.pick(loc_rand);
```

Comment:
This is not the least about auto, it's about pointers vs references. My example only had auto as I was to lazy to type the types.

> figure out the right syntax to use operator[] on a pointer to a vector

`x = (*some_pointer)[index];` or `x = some_pointer->operator[](index);`